src/producer/index.js

```diff
@@ -1,8 +1,17 @@
 const createRetry = require('../retry')
 const createDefaultPartitioner = require('./partitioners/default')
 const createSendMessages = require('./sendMessages')
+const InstrumentationEventEmitter = require('../instrumentation/emitter')
+const events = require('./instrumentationEvents')
+const { CONNECT, DISCONNECT } = require('./instrumentationEvents')
 const { KafkaJSNonRetriableError } = require('../errors')
 
+const { values, keys } = Object
+const eventNames = values(events)
+const eventKeys = keys(events)
+  .map(key => `consumer.events.${key}`)
+  .join(', ')
```

Comment:
Same here, it should be `producer`

Reply:
And this, kids, is why we try not to copy & paste code 😅I was considering breaking out some of the shared code between consumer, producer and admin, but it felt a bit premature. I'll just fix the copypaste errors first.

Reply:
:smile:

Reply:
Turns out the tests for this were passing for the wrong reason. `toThrow` only takes a single argument, which can be either a constructor, in which case it checks that the error is an instance of that, or a string or regex, in which case it matches on the error message. The tests were trying to do both, using `.toThrow(KafkaJSNonRetriableError, /blah blah consumer.events./)`, but since the regex was never taken into consideration, they were passing even if the event names were wrong.

Jest doesn't actually have a way of testing both, so for now I'm just testing the message, and not the error type. To test both, we'd have to write a custom matcher.